cmd/registry/patch/yaml.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package patch
+
+import (
+	"bytes"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Prefer this to yaml.Marshal because it uses tighter 2-space indentation.
+func marshalYAML(a interface{}) ([]byte, error) {
```

Comment:
Can we reduce the responsibility of this helper? IMO the important part is that we use a common encoder. We only need a unified encoding format, not an abstraction of the `yaml` package's interface.

```
func yamlEncoder(dst io.Writer) *yaml.Encoder {
	enc := yaml.NewEncoder(dst)
	enc.SetIndent(2)
	return enc
} 
```

Reply:
OK! done